.githooks/pre-commit

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env bash
+#
+# Pre-commit hook to check if CHANGELOG.md has been included
+# when the project root or relevant module has been edited.
+#
+# If it has been included, the latest version is compared
+# against the previous version by stripping all leading zeros
+# and comparing them. If it is greater it goes on to check
+# that the latest change's date is the current date (i.e. today).
+# If any of these tests fail the script exits and cancels the
+# commit.
+#
+# To bypass this validation add the `--no-verify` or `-n` flag:
+# 		`git commit --no-verify ...`
+
+# Get names of staged files being committed. If being run as
+# a git hook, take the result of git diff. If not, receive
+# the edited files from the CLI arguments provided.
+if [[ "${#}" -eq 0 ]]
+then
+    STAGED_FILES=($(git diff --cached --name-only)) # Local git hook
+else
+    STAGED_FILES=("${@}") # Github workflow passes arguments
+fi
+
+if [[ "${#STAGED_FILES[@]}" == 0 ]]
+then
+    echo "No files modified"
+    exit 0
+fi
+
+ROOT_EDITED=0
+ROOT_CHANGELOG_INCLUDED=0
```

Comment:
Do we even have a root changelog in the current repo?

Reply:
There is no root `CHANGELOG.md`